analyzers/src/SonarAnalyzer.Common/Rules/DebuggerDisplayUsesExistingMembersBase.cs

```diff
@@ -0,0 +1,89 @@
+    private readonly Regex evaluatedExpressionRegex = new(@"\{(?<EvaluatedExpression>[^\}]+)\}", RegexOptions.Multiline, RegexConstants.DefaultTimeout);
+
+    protected abstract SyntaxNode GetAttributeFormatString(TAttributeSyntax attribute);
+    protected abstract bool IsValidMemberName(string memberName);
+
+    protected override string MessageFormat => "'{0}' doesn't exist in this context.";
+
+    protected DebuggerDisplayUsesExistingMembersBase() : base(DiagnosticId) { }
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterNodeAction(Language.GeneratedCodeRecognizer,
+            c =>
+            {
+                var attribute = (TAttributeSyntax)c.Node;
+                if (Language.Syntax.IsKnownAttribute(attribute, KnownType.System_Diagnostics_DebuggerDisplayAttribute, c.SemanticModel)
+                    && GetAttributeFormatString(attribute) is { } formatString
+                    && FirstInvalidMemberName(c, formatString.GetFirstToken().ValueText, attribute) is { } firstInvalidMember)
+                {
+                    c.ReportIssue(Diagnostic.Create(Rule, formatString.GetLocation(), firstInvalidMember));
+                }
+            },
+            Language.SyntaxKind.Attribute);
+
+    private string FirstInvalidMemberName(SonarSyntaxNodeReportingContext context, string formatString, TAttributeSyntax attributeSyntax)
+    {
+        try
+        {
+            foreach (Match match in evaluatedExpressionRegex.Matches(formatString))
+            {
+                if (match.Groups["EvaluatedExpression"] is { Success: true, Value: var evaluatedExpression }
+                    && ExtractValidMemberName(evaluatedExpression) is { } memberName
+                    && GetAttributeTarget(attributeSyntax) is { } targetSyntax
+                    && context.SemanticModel.GetDeclaredSymbol(targetSyntax) is { } targetSymbol
+                    && GetTypeContainingReferencedMembers(targetSymbol) is { } typeSymbol
+                    && typeSymbol.GetSelfAndBaseTypes().SelectMany(x => x.GetMembers()).All(x => Language.NameComparer.Compare(x.Name, memberName) != 0))
+                {
+                    return memberName;
+                }
+            }
+
+            return null;
+        }
+        catch (RegexMatchTimeoutException)
+        {
+            return null;
+        }
+
+        string ExtractValidMemberName(string evaluatedExpression)
+        {
+            var sanitizedExpression = noQuotesModifierExpressionRegex.Replace(evaluatedExpression, string.Empty).Trim();
+            return IsValidMemberName(sanitizedExpression) ? sanitizedExpression : null;
+        }
+
+        static SyntaxNode GetAttributeTarget(TAttributeSyntax attributeSyntax) => attributeSyntax.Parent?.Parent;
+
+        static ITypeSymbol GetTypeContainingReferencedMembers(ISymbol symbol) => symbol is ITypeSymbol typeSymbol ? typeSymbol : symbol.ContainingType;
```

Comment:
Needs a new line, not a trivial one.
Doesn't need `Get` prefix
```suggestion
        static ITypeSymbol TypeContainingReferencedMembers(ISymbol symbol) => 
            symbol is ITypeSymbol typeSymbol ? typeSymbol : symbol.ContainingType;
```